internal/extproc/translator/openai_awsbedrock.go

```diff
@@ -152,113 +180,194 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIToolsToBedrockToolC
 	return nil
 }
 
-// OpenAIMessageToBedrockMessage converts openai ChatCompletion messages to aws bedrock messages
-func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) OpenAIMessageToBedrockMessage(openAIReq *openai.ChatCompletionRequest,
+// openAIMessageToBedrockMessageRoleUser converts openai user role message
+// returns a list of bedrock message and error
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleUser(
+	openAiMessage *openai.ChatCompletionMessageParamUnion,
+) ([]*awsbedrock.Message, error) {
+	message := openAiMessage.Value.(openai.ChatCompletionUserMessageParam)
+	bedrockMessages := make([]*awsbedrock.Message, 0)
+
+	if _, ok := message.Content.Value.(string); ok {
+		bedrockMessages = append(bedrockMessages, &awsbedrock.Message{
+			Role: openAiMessage.Type,
+			Content: []*awsbedrock.ContentBlock{
+				{Text: ptr.To(message.Content.Value.(string))},
+			},
+		})
+	} else if contents, ok := message.Content.Value.([]openai.ChatCompletionContentPartUserUnionParam); ok {
+		chatMessage := &awsbedrock.Message{Role: openAiMessage.Type}
+		chatMessage.Content = make([]*awsbedrock.ContentBlock, 0, len(contents))
+		for _, contentPart := range contents {
+			if contentPart.TextContent != nil {
+				textContentPart := contentPart.TextContent
+				chatMessage.Content = append(chatMessage.Content, &awsbedrock.ContentBlock{
+					Text: &textContentPart.Text,
+				})
+			} else if contentPart.ImageContent != nil {
+				imageContentPart := contentPart.ImageContent
+				parts := strings.Split(imageContentPart.ImageURL.URL, ",")
+				if len(parts) == 2 {
+					formatPart := strings.Split(parts[0], ";")[0]
+					format := strings.TrimPrefix(formatPart, "data:image/")
```

Comment:
also this really needs a comment

Reply:
I will update the how we parse data uri and add comments